zuul-core/src/test/java/com/netflix/zuul/netty/server/ClientRequestReceiverTest.java

```diff
@@ -46,23 +48,26 @@
 public class ClientRequestReceiverTest {
 
     @Test
-    public void proxyProtocol_portSetInContext() {
+    public void proxyProtocol_portSetInContextAndInHttpRequestMessage() {
         HAProxyMessage hapm = new HAProxyMessage(
                 HAProxyProtocolVersion.V2,
                 HAProxyCommand.PROXY,
                 HAProxyProxiedProtocol.TCP4,
                 "1.1.1.1", "2.2.2.2", 9000, 444);
-        ClientRequestReceiver receiver = new ClientRequestReceiver(null);
-        EmbeddedChannel channel = new EmbeddedChannel(receiver);
+        InetSocketAddress hapmDestinationAddress = new InetSocketAddress(InetAddresses.forString("2.2.2.2"), 444);
+        EmbeddedChannel channel = new EmbeddedChannel(new ClientRequestReceiver(null));
         channel.attr(SourceAddressChannelHandler.ATTR_SERVER_LOCAL_PORT).set(1234);
-        channel.attr(HAProxyMessageChannelHandler.ATTR_HAPROXY_MESSAGE).set(hapm);
+        channel.attr(SourceAddressChannelHandler.ATTR_PROXY_PROTOCOL_DESTINATION_ADDRESS).set(hapmDestinationAddress);
         HttpRequestMessageImpl result;
         {
             channel.writeInbound(new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, "/post", Unpooled.buffer()));
             result = channel.readInbound();
             result.disposeBufferedBody();
         }
-        assertEquals(result.getContext().get(CommonContextKeys.PROXY_PROTOCOL_PORT), 444);
+        // not sure how to wire this next assertion up, SourceAddressChannelHandler is needed in the channel pipeline
```

Comment:
@argha-c how can I wire `SourceAddressChannelHandler` in so I can perform the assertion on the next line?

Reply:
You'd need to explicitly add the handler to the channel pipeline, if you're relying on that to fire.
It seems, in your case, you need the `HAProxyMessageChannelHandler` to read though, not `SourceAddressChannelHandler`?